src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -454,7 +454,11 @@ InstanceKlass* InstanceKlass::allocate_instance_klass(const ClassFileParser& par
   assert(loader_data != nullptr, "invariant");
 
   InstanceKlass* ik;
-  const bool use_class_space = !parser.is_interface() && !parser.is_abstract();
+  // Only Klasses that aren't interface or abstract classes are found in oop headers so they
+  // are the only ones that require compression and allocation in the limited class metaspace
+  // region.  But ... there are C2 dependencies that all klasses are allocated to the class metaspace, so
+  // always use the class metaspace until JDK-NNNNNNN is resolved.
```

Comment:
JDK-NNNNNN was supposed to be replaced by an actual bug number. :)

Reply:
I'm left wondering what this is supposed to do. We made changes, then we are disabling them until something is fixed but we are not saying what something that is. ???

Reply:
Given that it's a subtask of the bug that we want to fix, it's pretty easy to see with git blame.  Like with all changes, we don't put bug numbers in.  But the bug will be fixed soon so I'm closing this as WNF.